server/models/Document.js

```diff
@@ -556,6 +555,18 @@ Document.prototype.publish = async function (options) {
   return this;
 };
 
+Document.prototype.unpublish = async function (options) {
+  if (!this.publishedAt) return this.save(options);
```

Comment:
```suggestion
  if (!this.publishedAt) return this;
```

Can we just avoid the save here altogether?